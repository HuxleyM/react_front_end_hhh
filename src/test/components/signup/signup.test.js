import SignUp from '../../../components/signup.js'
import SignUpForm from '../../../components/signupForm.js'
import React, {Component} from 'react';
import { create } from "react-test-renderer";
import {shallow, mount, render} from 'enzyme'

describe("Sign up", () => {
  test('it Matches the snapshot', () => {
    const component = create(<SignUp/>);
    expect(component.toJSON()).toMatchSnapshot();
  })
})

describe("Sign up", () => {

  let wrapper
  let userinfo

  beforeEach(() => {
      userinfo = {
      name: 'Crisis Cafe',
      address: 'Commercial Street',
      email: 'c@cc.com',
      password: 'password123'
    }
    wrapper = mount(<SignUp/>,
    { attachTo: document.body })
  })

  it('renders a form (sanity check)', () =>{
    expect(wrapper.contains('Venue Name')).toBe(true)
  })

  it('Calls handleForm on submit click', () => {

    wrapper.instance().sendVenue = jest.fn()

    const venueNameInput = wrapper.find('input.venuename')
    venueNameInput.value = userinfo.name

    const venueAddressInput = wrapper.find('input.venueaddress')
    venueAddressInput.value = userinfo.address

    const venueEmailInput = wrapper.find('input.venueEmail')
    venueEmailInput.value = userinfo.email

    const venuePasswordInput = wrapper.find('input.password')
    venuePasswordInput.value = userinfo.password

    wrapper.find('input.submitForm').simulate('submit');

    expect(wrapper.instance().sendVenue).toHaveBeenCalled()

  })




})


// describe('<LogIn />', () => {
//     const testValues = {
//         username: 'FOO',
//         password: 'BAZ',
//         handleSubmit: jest.fn(),
//     };

//     it('Submit works', () => {

//         const component = shallow(
//             <LogIn {...testValues} />
//         );
//         component.find('#submitButton').simulate('click');
//         expect(testValues.handleSubmit).toHaveBeenCalledTimes(1);
//         expect(testValues.handleSubmit).toBeCalledWith({username: testValues.username, password: testValues.password});
//     });
// });
