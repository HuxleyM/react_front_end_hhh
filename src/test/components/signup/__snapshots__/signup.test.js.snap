// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sign up it Matches the snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <label>
      Venue Name
    </label>
    <input
      id="venuename"
      placeholder="Crisis Cafe"
      required={true}
      type="text"
    />
    <label>
      Venue Address
    </label>
    <input
      id="venueaddress"
      placeholder="10 Commercial Street"
      required={true}
      type="text"
    />
    <label>
      Email
    </label>
    <input
      id="venueEmail"
      placeholder="venue@gmail.com"
      required={true}
      type="email"
    />
    <label>
      Password
    </label>
    <input
      id="password"
      minLength="6"
      placeholder="qwe123"
      required={true}
      type="password"
    />
    <input
      className="submitForm action_button"
      id="submit_form"
      type="submit"
      value="Submit"
    />
  </form>
</div>
`;
